js/ui/animation.js

```diff
@@ -63,7 +63,16 @@ var Animation = new Lang.Class({
 
     _update: function() {
         this._showFrame(this._frame + 1);
-        return GLib.SOURCE_CONTINUE;
+
+        // Show the next frame after the timeout of the current one
+        let currentFrameTimeout = this._frame % this._frameTimeouts.length;
```

Comment:
This variable is an index, not the timeout itself

Reply:
I know. Anyway, I will add the `Index` suffix if that makes it more clear.